clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -238,6 +240,15 @@ data WireOrReg = Wire | Reg
 
 instance NFData WireOrReg
 
+pattern InstDecl
+  :: EntityOrComponent
+  -> Maybe Identifier
+  -> Identifier
+  -> Identifier
+  -> [(Expr,PortDirection,HWType,Expr)]
+  -> Declaration
+pattern InstDecl eoc mi i2 i3 as = InstDecl' eoc mi i2 i3 [] as
```

Comment:
Is there a good reason to go with patterns here? As far as I can see, it adds an layer of abstraction with very little benefit. `InstDecl` is only used in ~10 locations, maybe you could just change those lines to include `[]`?

Reply:
I did it this way because it's similar to how `InstDecl` is done. Also, even though this stuff is "internal", making this change would likely break code for anyone using the haskell blackbox api, which I wanted to avoid. But I'm not too fussed either way.